cpu/sam0_common/periph/uart.c

```diff
@@ -317,6 +317,44 @@ void uart_poweroff(uart_t uart)
     sercom_clk_dis(dev(uart));
 }
 
+#ifdef MODULE_PERIPH_UART_COLLISION
+bool uart_collision_detected(uart_t uart)
+{
+    bool collision = dev(uart)->STATUS.bit.COLL;
+    dev(uart)->STATUS.reg = SERCOM_USART_STATUS_COLL;
+    return collision;
+}
+
+void uart_collision_detect_enable(uart_t uart)
+{
+    /* clear stale collision flag */
+    dev(uart)->STATUS.reg = SERCOM_USART_STATUS_COLL;
+
+    /* enable collision detection */
+    dev(uart)->CTRLB.bit.COLDEN = 1;
+
```

Comment:
You might want to wait for the CTRLB sync bit here: `while (dev(uart)->SYNCBUSY.bit.CTRLB) { }`